pkgs/by-name/ut/util-linux/package.nix

```diff
@@ -113,17 +128,21 @@ stdenv.mkDerivation rec {
     "usrsbin_execdir=${placeholder "bin"}/sbin"
   ];
 
-  nativeBuildInputs = [
-    pkg-config
-    installShellFiles
-  ] ++ lib.optionals translateManpages [ po4a ];
+  nativeBuildInputs =
+    [
+      pkg-config
+      installShellFiles
+    ]
+    ++ lib.optionals translateManpages [ po4a ]
+    ++ lib.optionals (cryptsetupSupport == "dlopen") [ cryptsetup ];
 
   buildInputs =
     [
       zlib
       libxcrypt
       sqlite
     ]
+    ++ lib.optionals (cryptsetupSupport == true) [ cryptsetup ]
```

Comment:
Could use `lib.optional`

Reply:
All the surrounding lines were using `optionals` so stuck with that.